datumaro/plugins/yolo_format/extractor.py

```diff
@@ -334,25 +334,28 @@ def _config(self) -> Dict[str, Any]:
             except yaml.YAMLError:
                 raise InvalidAnnotationError("Failed to parse config file")
 
-    def _load_categories(self) -> Dict[AnnotationType, LabelCategories]:
+    def _load_names_from_config_file(self):
+        names = self._config["names"]
+        if isinstance(names, dict):
+            if set(names.keys()) != set(range(len(names))):
+                raise InvalidAnnotationError(
+                    "Failed to parse names from config: non-sequential label ids"
+                )
```

Comment:
What is the point of this check? If it's a dict, I think it should support arbitrary ordering.

Reply:
YOLOv8 framework requires `n` class dataset to have class indices `0` - `n - 1`. 
https://github.com/ultralytics/ultralytics/blob/main/ultralytics/nn/autobackend.py#L36

Do you think that datumaro should be able to import a dataset even if it does not meet the YOLOv8 requirements?

Reply:
Done

Reply:
btw, may be we should also allow for string keys?

Reply:
Do you mean arbitrary strings or strings with numbers? The latter one seems to be supported in the code by the link above, so it makes sense to support it in Datumaro as well.

Reply:
I mean arbitrary strings

Reply:
Could you add an example of the input and supposed parsing result?

Reply:
data.yaml has something like that:
```yaml
names:
  0: label0 # <- this is what YOLOv8 supports 
  1: label1  # <- this is what YOLOv8 supports 
  42: another_label  # <- this is allowed by this PR
  string_key: yet_another_label   # <- I suggest to also allow this
```
and then annotation txt file could have something like
```
0 <coordinates>
42 <coordinates>
string_key <coordinates>
```

Keys are dropped while parsing, therefore it is supposed to be parsed to the same dataset as if data.yaml had
```yaml
names:
  0: label0 
  1: label1
  2: another_label 
  3: yet_another_label
```
with annotation txt file
```
0 <coordinates>
2 <coordinates>
3 <coordinates>
```